modyn/models/fmownet/fmownet.py

```diff
@@ -0,0 +1,29 @@
+from typing import Any
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+from torchvision.models import densenet121
+
+
+class FmowNet:
+    # pylint: disable-next=unused-argument
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        self.model = FmowNetModel(**model_configuration)
+        self.model.to(device)
+
+
+class FmowNetModel(nn.Module):
+    def __init__(self, num_classes: int) -> None:
+        super().__init__()
+        self.num_classes = num_classes
+        self.enc = densenet121(pretrained=True).features
+        self.classifier = nn.Linear(1024, self.num_classes)
```

Comment:
is there no need for a softmax?

Reply:
Softmax is tipically hidden inside the CrossEntropy loss function. See [this discussion](https://discuss.pytorch.org/t/softmax-cross-entropy-loss/125383) for example

Reply:
Is all hidden inside the densenet. In total, FMoWNet has 7007156 parameters